test/xpu_api/utilities/optional/optional.object/optional.object.ctor/rvalue_T.pass.cpp

```diff
@@ -0,0 +1,83 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/optional>
+#include <oneapi/dpl/type_traits>
+#include <oneapi/dpl/utility>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+using dpl::optional;
+
+bool
+kernel_test()
+{
+    sycl::queue q = TestUtils::get_test_queue();
+    bool ret = true;
+    sycl::range<1> numOfItems1{1};
+    {
+        sycl::buffer<bool, 1> buffer1(&ret, numOfItems1);
+
+        q.submit([&](sycl::handler& cgh) {
+            auto ret_access = buffer1.get_access<sycl::access::mode::write>(cgh);
+            cgh.single_task<class KernelTest>([=]() {
+                {
+                    typedef int T;
+                    constexpr optional<T> opt(T(5));
+                    static_assert(static_cast<bool>(opt) == true);
+                    static_assert(*opt == 5);
+
+                    struct test_constexpr_ctor : public optional<T>
+                    {
+                        constexpr test_constexpr_ctor(T&&) {}
```

Comment:
It actually tests if the default constructor of `optional` is constexpr. This constructor should call the right constructor of the base class.

Also, I don't see much value in this struct. If the constructor of optional is not constexpr, the test on line 42 will fail.

Reply:
Fixed